From 9025ffe9ebe6dcb1b47d5024a6a09ee9461a12a6 Mon Sep 17 00:00:00 2001
From: Jason Wang <jasowang@redhat.com>
Date: Thu, 9 Oct 2014 06:13:25 -0400
Subject: [virt] hyperv/vmbus: Fix a bug in vmbus_open()

Message-id: <1412835210-22435-7-git-send-email-jasowang@redhat.com>
Patchwork-id: 97288
O-Subject: [RHEL7.1 PATCH V2 06/11] Drivers: hv: vmbus: Fix a bug in vmbus_open()
Bugzilla: 1146358
RH-Acked-by: Vitaly Kuznetsov <vkuznets@redhat.com>

From: K. Y. Srinivasan <kys@microsoft.com>

Bugzilla: 1146358

Fix a bug in vmbus_open() and properly propagate the error. I would
like to thank Dexuan Cui <decui@microsoft.com> for identifying the
issue.

Signed-off-by: K. Y. Srinivasan <kys@microsoft.com>
Cc: <stable@vger.kernel.org>
Tested-by: Sitsofe Wheeler <sitsofe@yahoo.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
(cherry picked from commit 45d727cee9e200f5b351528b9fb063b69cf702c8)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/hv/channel.c b/drivers/hv/channel.c
index 0303a1b..80ba165 100644
--- a/drivers/hv/channel.c
+++ b/drivers/hv/channel.c
@@ -208,8 +208,10 @@ int vmbus_open(struct vmbus_channel *newchannel, u32 send_ringbuffer_size,
  ret = vmbus_post_msg(open_msg,
           sizeof(struct vmbus_channel_open_channel));
 
- if (ret != 0)
+ if (ret != 0) {
+  err = ret;
   goto error1;
+ }
 
  t = wait_for_completion_timeout(&open_info->waitevent, 5*HZ);
  if (t == 0) {
-- 
1.7.1